x-pack/filebeat/input/netflow/netflow_test.go

```diff
@@ -60,6 +67,108 @@ type TestResult struct {
 	Flows []beat.Event `json:"events,omitempty"`
 }
 
+func newV2Context() (v2.Context, func()) {
+	ctx, cancel := context.WithCancel(context.Background())
+	return v2.Context{
+		Logger:      logp.NewLogger("netflow_test"),
+		ID:          "test_id",
+		Cancelation: ctx,
+	}, cancel
+}
+
+func TestNetFlow(t *testing.T) {
+	pcaps, err := filepath.Glob(filepath.Join(pcapDir, "*.pcap"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, file := range pcaps {
+		testName := strings.TrimSuffix(filepath.Base(file), ".pcap")
+
+		t.Run(testName, func(t *testing.T) {
+
+			pluginCfg, err := conf.NewConfigFrom(mapstr.M{})
+			require.NoError(t, err)
+
+			netflowPlugin, err := Plugin(logp.NewLogger("netflow_test")).Manager.Create(pluginCfg)
+			require.NoError(t, err)
+
+			mockPipeline := &pipelinemock.MockPipelineConnector{}
+
+			ctx, cancelFn := newV2Context()
+			errChan := make(chan error)
+			go func() {
+				defer close(errChan)
+				errChan <- netflowPlugin.Run(ctx, mockPipeline)
+			}()
+
+			defer cancelFn()
+
+			require.Eventually(t, mockPipeline.HasConnectedClients, 5*time.Second, 100*time.Millisecond,
+				"client connects to pipeline")
```

Comment:
It's not a blocker, but I believe we could improve the message here. "client connects to pipeline" is what is going to apper in the test failure output:
```
--- FAIL: TestNetFlow (3.00s)
    --- FAIL: TestNetFlow/ipfix_cisco (1.00s)
        netflow_test.go:107: 
                Error Trace:    /home/tiago/devel/beats/x-pack/filebeat/input/netflow/netflow_test.go:107
                Error:          Condition never satisfied
                Test:           TestNetFlow/ipfix_cisco
                Messages:       client connects to pipeline
```

I believe it could be more clear about why it failed. Something like:
```suggestion
			require.Eventually(t, mockPipeline.HasConnectedClients, 5*time.Second, 100*time.Millisecond,
				"no client has connected to the pipeline")
```

Reply:
@belimawr please re-review after the above changes